<a href="https://colab.research.google.com/github/Daarwinmendez/heart_disease_classifier/blob/main/heart_disease_classifier.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import sklearn as sk
from sklearn.ensemble import RandomForestClassifier
import sklearn.model_selection as ms
import numpy as np
import pandas as pd

data = pd.read_csv("https://raw.githubusercontent.com/mrdbourke/zero-to-mastery-ml/master/data/heart-disease.csv")
# Table was taken from Daniel Bourke's GitHub


data.head() # Just brought five rows, but there are plenty of them (303) in total, and 14 columns.

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   63    1   3       145   233    1        0      150      0      2.3      0   
1   37    1   2       130   250    0        1      187      0      3.5      0   
2   41    0   1       130   204    0        0      172      0      1.4      2   
3   56    1   1       120   236    0        1      178      0      0.8      2   
4   57    0   0       120   354    0        1      163      1      0.6      2   

   ca  thal  target  
0   0     1       1  
1   0     2       1  
2   0     2       1  
3   0     2       1  
4   0     2       1

In [ ]:
# Let's confirm the shape.
data.shape

(303, 14)

#### Those column titles up there may seem chinese if i don't explain them.

So let's break certain things things down, down below.

These are what they mean:

1. **age**: Age - The age of the individual.
2. **sex**: Sex - The gender of the individual (1 = male, 0 = female).
3. **cp**: Chest Pain Type - The type of chest pain experienced (0-3).
4. **trestbps**: Resting Blood Pressure - The resting blood pressure in mm Hg.
5. **chol**: Serum Cholesterol - The serum cholesterol level in mg/dL.
6. **fbs**: Fasting Blood Sugar - Fasting blood sugar level > 120 mg/dL (1 = true, 0 = false).
7. **restecg**: Resting Electrocardiographic Results - The results of the resting electrocardiogram (0-2).
8. **thalach**: Maximum Heart Rate Achieved - The highest heart rate achieved during the test.
9. **exang**: Exercise Induced Angina - Whether exercise-induced angina is present (1 = yes, 0 = no).
10. **oldpeak**: ST Depression - The ST depression induced by exercise relative to rest.
11. **slope**: Slope of Peak Exercise ST Segment - The slope of the peak exercise ST segment (0-2).
12. **ca**: Number of Major Vessels - The number of major vessels colored by fluoroscopy (0-3).
13. **thal**: Thalassemia - A blood disorder type (1 = normal, 2 = fixed defect, 3 = reversible defect).
14. **target**: Target - The presence of heart disease (1 = yes, 0 = no).


In [ ]:
# First we set the random seed to 42
np.random.seed(42)

# Let's obtain 'X' & 'y'
X = data.drop("target", axis=1)
y =  data["target"]

# Initialize the RandomForestClassifier model
model = RandomForestClassifier(n_estimators=100);

# Split the data into training and testing sets, 80% for training, 20% for test.
X_train, X_test, y_train, y_test = ms.train_test_split(X,y, train_size=0.8, test_size=0.2);

# Train the model
model.fit(X_train, y_train);

In [ ]:
# Print the accuracy of the model on the test data
print(model.score(X_test, y_test));

0.8524590163934426


In [ ]:
# Let's predict the target value for the test data.
model.predict(X_test)


# One means: Patient has heart disease,
# Zero means: Patient does not have heart disease.

array([0, 1, 1, 0, 1, 1, 1, 0, 0, 1, 1, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0,
       1, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1,
       1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0])

#### A pleasure *to* present this to y'all

<img alt="Libraries Images" style="width: 100%; height: auto;" src="https://drive.google.com/uc?export=view&id=1XD4rWCqQ3ujO5Wjx3ZVQLddEF2Mq8a6h">
